projects/offender-events-and-delius/src/test/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/offender/OffenderDeltaServiceTest.kt

```diff
@@ -0,0 +1,782 @@
+package uk.gov.justice.digital.hmpps.integrations.delius.offender
+
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.MethodSource
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.kotlin.*
+import uk.gov.justice.digital.hmpps.integrations.delius.person.entity.RegisterType
+import uk.gov.justice.digital.hmpps.integrations.delius.person.entity.Registration
+import uk.gov.justice.digital.hmpps.integrations.delius.person.entity.RegistrationRepository
+import uk.gov.justice.digital.hmpps.integrations.delius.referencedata.entity.ReferenceData
+import uk.gov.justice.digital.hmpps.integrations.delius.service.DomainEventService
+import uk.gov.justice.digital.hmpps.message.Notification
+import uk.gov.justice.digital.hmpps.publisher.NotificationPublisher
+import uk.gov.justice.digital.hmpps.telemetry.TelemetryService
+import java.time.ZonedDateTime
+import java.util.*
+
+@ExtendWith(MockitoExtension::class)
+class OffenderDeltaServiceTest {
+
+    @Mock
+    lateinit var offenderDeltaRepository: OffenderDeltaRepository
+
+    @Mock
+    lateinit var contactRepository: ContactRepository
+
+    @Mock
+    lateinit var notificationPublisher: NotificationPublisher
+
+    @Mock
+    lateinit var telemetryService: TelemetryService
+
+    @Mock
+    lateinit var domainEventService: DomainEventService
+
+    @Mock
+    lateinit var registrationRepository: RegistrationRepository
+
+    private lateinit var offenderDeltaService: OffenderDeltaService
+
+    @BeforeEach
+    fun setUp() {
+        offenderDeltaService = OffenderDeltaService(
+            batchSize = 5,
+            offenderDeltaRepository = offenderDeltaRepository,
+            contactRepository = contactRepository,
+            notificationPublisher = notificationPublisher,
+            telemetryService = telemetryService,
+            domainEventService = domainEventService,
+            registrationRepository = registrationRepository,
+        )
+    }
+
+    @ParameterizedTest
+    @MethodSource("nonContactTables")
+    fun `when sourceTable is not CONTACT then no domain event is generated`(table: String) {
+        // given
+        val offender = Offender(
+            id = 99L,
```

Comment:
Minor thing - given there are no assertions on specific IDs in this test class, you could replace all these random numbers with `id()` to make things a bit cleaner.